path_pattern/pattern.go

```diff
@@ -23,33 +24,45 @@ func (p *Pattern) UnmarshalText(data []byte) error {
 	return nil
 }
 
-// Match happens if the prefix of the input matches the pattern.
+func (p Pattern) regexp() *regexp.Regexp {
+	var pieces []string
+	for _, piece := range p {
+		pieces = append(pieces, piece.Regexp())
+	}
+	return regexp.MustCompile("^" + strings.Join(pieces, "/") + "$")
+}
+
+// Match happens if the pattern exactly matches the string.
 func (p Pattern) Match(v string) bool {
-	parts := strings.Split(v, "/")
+	r := p.regexp()
+	return r.MatchString(removeTrailingSlashes(v))
+}
 
-	compLen := len(p)
-	if compLen > len(parts) {
-		return false
-	}
+func (p Pattern) MatchWithGroup(v string) (bool, []string) {
```

Comment:
A comment documenting the returned `[]string` value (ideally with examples) would be useful.

Reply:
Fixed in 591c5e0.